File: flex/rgb-led/rgb-led.c
#include <wiringPi.h>
#include <stdio.h>
#include <signal.h>
#include <stdlib.h>

//Pin 33 on Raspberry Pi corresponds to BCM GPIO 13 and wiringPi pin 22
#define Pin0 23
//Pin 35 on Raspberry Pi corresponds to BCM GPIO 19 and wiringPi pin 24
#define Pin1 24
//Pin 37 on Raspberry Pi corresponds to BCM GPIO 26 and wiringPi pin 25
#define Pin2 25

void led(int red, int green, int blue)
{
  digitalWrite(Pin0, red);
  digitalWrite(Pin1, green);
  digitalWrite(Pin2, blue);
}

void close()
{
  printf("\nTurning off lights...\n");
  led(LOW, LOW, LOW);
  exit(0);
}

int main(void)
{
  signal(SIGINT, close);

  if(-1 == wiringPiSetup())
  {
      printf("setup wiringPi failed!");
      return 1;
  }

  //Set GPIO pins
  pinMode(Pin0, OUTPUT);
  pinMode(Pin1, OUTPUT);
  pinMode(Pin2, OUTPUT);

  //Play a sound until the user closes the app
  while(1)
  {
    led(HIGH, LOW, LOW);
    delay(3000);
    led(LOW, HIGH, LOW);
    delay(3000);
    led(LOW, LOW, HIGH);
    delay(3000);
  }

  return 0;
}

File: flex/rgb-led/python/rgb-led.py
# coding=utf-8
import signal
import sys
import time

import RPi.GPIO as GPIO

# Blue - pin 33 on Raspberry Pi corresponds to BCM GPIO 13
Pin0 = 33
# Green - pin 35 on Raspberry Pi corresponds to BCM GPIO 19
Pin1 = 35
# Red - pin 37 on Raspberry Pi corresponds to BCM GPIO 26
Pin2 = 37


def led(red, green, blue):
    GPIO.output(Pin0, blue)
    GPIO.output(Pin1, green)
    GPIO.output(Pin2, red)


def close(signal, frame):
    print("\nTurning off lights...\n")
    led(GPIO.LOW, GPIO.LOW, GPIO.LOW)
    sys.exit(0)


signal.signal(signal.SIGINT, close)

# to use Raspberry Pi board pin numbers
GPIO.setmode(GPIO.BOARD)
GPIO.setwarnings(False)

# set up GPIO output channel
GPIO.setup(Pin0, GPIO.OUT)
GPIO.setup(Pin1, GPIO.OUT)
GPIO.setup(Pin2, GPIO.OUT)

# reset
led(GPIO.LOW, GPIO.LOW, GPIO.LOW)

try:
    while True:
        led(GPIO.HIGH, GPIO.LOW, GPIO.LOW)
        time.sleep(3)
        led(GPIO.LOW, GPIO.HIGH, GPIO.LOW)
        time.sleep(3)
        led(GPIO.LOW, GPIO.LOW, GPIO.HIGH)
        time.sleep(3)
finally:
    GPIO.cleanup()
